src/components/Chat.astro

```diff
@@ -54,16 +73,55 @@ import Button from "../components/Button.astro";
     const assistantMessage =
       chatCompletion.choices[0]?.message?.content || "No response";
     chatHistory.push({ role: "assistant", content: assistantMessage });
-
     return assistantMessage;
   }
 
+  // Function to display messaging history
+  function displayMessages() {
+    const historyDiv = document.getElementById("history");
+    historyDiv.innerHTML = ""; // Clear previous messages
+    for (let i = 1; i < chatHistory.length; i++) {
+      const msg = chatHistory[i];
+      const messageDiv = document.createElement("div");
+      messageDiv.classList.add(
+        "p-2",
+        "px-4",
+        "rounded-3xl",
+        "max-w-xs",
+        "mb-2"
+      );
+      if (msg.role === "user") {
+        messageDiv.classList.add(
+          "border",
+          "border-black",
+          "self-end",
+          "text-black"
+        );
+      } else if (msg.role === "assistant") {
+        messageDiv.classList.add("bg-black", "self-start", "text-off-white");
+      }
+      messageDiv.textContent = msg.content;
+      historyDiv.appendChild(messageDiv);
+    }
+  }
+
+  async function enterMessage() {
+    const user = document.getElementById("user") as HTMLInputElement;
+    if (user) {
+      const message = user.value;
+      document.getElementById("user").value = "";
+      await handleUserMessage(message);
+    }
+    displayMessages();
+  }
+
   document
     .getElementById("submit-button")
-    .addEventListener("click", async () => {
-      const user = (document.getElementById("user") as HTMLInputElement).value;
-      const message = await handleUserMessage(user);
-      document.getElementById("chat-output").textContent = message;
-      //document.getElementById("history").textContent = chatHistory.map((msg) => msg.content).join("\n");
-    });
+    .addEventListener("click", enterMessage);
+
+  document.getElementById("user").addEventListener("keydown", function (event) {
```

Comment:
Would it work to have `document.getElementById("user")` be called as a global var and then just reference it the other two times but maybe not?